src/SensorApplication.java

```diff
@@ -0,0 +1,154 @@
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.TitledBorder;
+
+public class SensorApplication extends JFrame {
+	
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+
+	public SensorApplication() {
+		setTitle("Sensor Tracker");
+		setLayout(new GridLayout(4,1));
+		
+		SensorAdapter sA = new SensorAdapter();
+		
+		//Pressure
+		JPanel  pressurePnl = new JPanel();
+		pressurePnl.setLayout(new GridLayout(2,3));
+		JLabel pressureLevel = new JLabel();
+		JPanel Pressure[] = new JPanel[6];
+		pressurePnl.setBorder(new TitledBorder(sA.getPressureName()));
+		add(pressurePnl);
+		for(int i=0;i<6;i++) {
+			Pressure[i]=new JPanel();
+			if(i==4)
+				pressurePnl.add(pressureLevel);
+			else
+				pressurePnl.add(Pressure[i]);
+		}
+		
+		//Radiation 
+		JPanel  radiationPnl = new JPanel();
+		radiationPnl.setLayout(new GridLayout(2,3));
+		JLabel radiationLevel = new JLabel();
+		JPanel Radiation[] = new JPanel[6];
+		radiationPnl.setBorder(new TitledBorder(sA.getRadiationName()));
+		add(radiationPnl);
+		for(int i=0;i<6;i++) {
+			Radiation[i]=new JPanel();
+			if(i==4)
+				radiationPnl.add(radiationLevel);
+			else
+				radiationPnl.add(Radiation[i]);
+		}
+		
+		//Temperature
+		JPanel  temperaturePnl = new JPanel();
+		temperaturePnl.setLayout(new GridLayout(2,3));
+		JLabel temperatureLevel = new JLabel();
+		JPanel Temperature[] = new JPanel[6];
+		temperaturePnl.setBorder(new TitledBorder(sA.getTemperatureName()));
+		add(temperaturePnl);
+		for(int i=0;i<6;i++) {
+			Temperature[i]=new JPanel();
+			if(i==4)
+				temperaturePnl.add(temperatureLevel);
+			else
+				temperaturePnl.add(Temperature[i]);
+		}
+		
+		JButton runTest = new JButton("Run Sensor Diagnostics");
```

Comment:
I really like what you've done here by implementing a button to run the diagnostics test. This saves the user time by eliminating the run process found in the users API.